typing/typemod.ml

```diff
@@ -3401,15 +3411,6 @@ and type_structure ?(toplevel = None) funct_body anchor env ?expected_mode
             (fun () -> Typecore.type_representable_expression
                          ~why:Structure_item_expression env sexpr)
         in
-        if force_toplevel then
-          (* See comment on [force_toplevel]. *)
-          begin match Jkind.Sort.default_to_value_and_get sort with
-          | Base Value -> ()
-          | Product _
-          | Base (Void | Untagged_immediate | Float64 | Float32 | Word |
-                 Bits8 | Bits16 | Bits32 | Bits64 | Vec128 | Vec256 | Vec512) ->
-            raise (Error (sexpr.pexp_loc, env, Toplevel_unnamed_nonvalue sort))
```

Comment:
I think the definition and printing of the `Toplevel_unnamed_nonvalue` error is now dead code and can be removed from this module.